sql/src/main/java/io/crate/executor/transport/NodeAction.java

```diff
@@ -30,4 +30,5 @@
     String actionName();
     String executorName();
     void nodeOperation(TRequest request, ActionListener<TResponse> listener);
+    void execute(String id, TRequest transportRequest, ActionListener<TResponse> actionListener);
```

Comment:
not sure if this really fits into the NodeAction interface because execute is executed on a different node than the other 3 operations. Imho who ever calls execute can use a concrete implementation (TransportKillJobsNodeAction in this case) or we should have a separate interface that has only this method.